tests/metrics/column_aggregate/test_column_aggregate.py

```diff
@@ -0,0 +1,91 @@
+from pathlib import Path
+
+import pandas
+import pytest
+from pytest import FixtureRequest
+
+from great_expectations.metrics.column_aggregate.mean import (
+    ColumnValuesMean,
+    ColumnValuesMeanResult,
+)
+from great_expectations.metrics.metric_results import MetricErrorResult
+from tests.integration.test_utils.data_source_config import (
+    PandasDataFrameDatasourceTestConfig,
+    PostgreSQLDatasourceTestConfig,
+    SparkFilesystemCsvDatasourceTestConfig,
+)
+from tests.integration.test_utils.data_source_config.pandas_data_frame import (
+    PandasDataFrameBatchTestSetup,
+)
+from tests.integration.test_utils.data_source_config.postgres import PostgresBatchTestSetup
+from tests.integration.test_utils.data_source_config.spark_filesystem_csv import (
+    SparkFilesystemCsvBatchTestSetup,
+)
+
+
+@pytest.fixture
+def dataframe() -> pandas.DataFrame:
+    return pandas.DataFrame(
+        {
+            "id": [1, 2, 3, 4],
+            "number": [1, 2, 3, 4],
+            "string": ["a", "b", "c", "d"],
+        },
+    )
+
+
+@pytest.fixture
+def setup_pandas(dataframe: pandas.DataFrame) -> PandasDataFrameBatchTestSetup:
+    return PandasDataFrameBatchTestSetup(
+        config=PandasDataFrameDatasourceTestConfig(),
+        data=dataframe,
+    )
+
+
+@pytest.fixture
+def setup_spark(dataframe: pandas.DataFrame, tmp_path: Path) -> SparkFilesystemCsvBatchTestSetup:
+    return SparkFilesystemCsvBatchTestSetup(
+        config=SparkFilesystemCsvDatasourceTestConfig(),
+        data=dataframe,
+        base_dir=tmp_path,
+    )
```

Comment:
You may need to pass `column_types` to `SparkFilesystemCsvDatasourceTestConfig` to get the spark error test passing. Maybe it is coercing somehow, I'm not sure why it doesn't fail.

Reply:
It looks to be a bug and we use a different metric name when the mean fails on spark. See inline comment on failure test below.